rfcs/button-component.md

```diff
@@ -0,0 +1,81 @@
+# Button Component
+
+Rough DOM composition:
+
+```
++-------------------------------------------------------------------+
+| BaseButton (display: flex; justifyContent: space-between)         |
+| +-----------------+-----------------------+---------------------+ |
+| |  startEnhancer  |         label         |     endEnhancer     | |
+| |  display:flex   |                       |     display:flex    | |
+| |                 |                       |                     | |
+| |                 |                       |                     | |
+| |                 |                       |                     | |
+| +-----------------+-----------------------+---------------------+ |
++-------------------------------------------------------------------+
+```
+
+### Exports
+
+* `Button`
+* `StyledButton`
+* `StyledButtonLabel`
+* `StyledStartEnhancer`
+* `StyledEndEnhancer`
+* `KIND`
+* `SIZE`
+* `SHAPE`
+
+### <Button/> API
+
+* `label: node | function` - Optional
+  A label in the button.
```

Comment:
What's the thought behind using `label` instead of just using `children`? It seems like material, antd, and native HTML buttons all use `children`, should we just stick with that?